miden-lib/src/notes/mod.rs

```diff
@@ -88,3 +115,26 @@ pub fn notes_try_from_elements(elements: &[Word]) -> Result<NoteStub, NoteError>
 
     Ok(stub)
 }
+
+/// Utility function generating RECIPIENT for the P2ID note script created by the ASWAP script
+fn build_p2id_recipient(target: AccountId, serial_num: Word) -> Result<Digest, NoteError> {
+    let assembler = assembler();
+
+    let p2id_bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/P2ID.masb"));
+
+    let note_script_ast =
+        ProgramAst::from_bytes(p2id_bytes).map_err(NoteError::NoteDeserializationError)?;
+
+    let (note_script, _) = NoteScript::new(note_script_ast, &assembler)?;
```

Comment:
Let's add a `TODO` here to investigate lazy static approach and also create an issue for this.